examples/distro/custom/src/main/java/com/example/javaagent/DemoResourceProvider.java

```diff
@@ -1,11 +1,29 @@
 package com.example.javaagent;
 
 import io.opentelemetry.api.common.Attributes;
-import io.opentelemetry.sdk.resources.ResourceProvider;
+import io.opentelemetry.sdk.autoconfigure.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+
+public class DemoResourceProvider implements ResourceProvider {
 
-public class DemoResourceProvider extends ResourceProvider {
   @Override
-  protected Attributes getAttributes() {
-    return Attributes.builder().put("custom.resource", "demo").build();
+  public Resource createResource(ConfigProperties config) {
+    return DemoResource.get();
+  }
+
+  private static final class DemoResource {
```

Comment:
Why do we need separate class? Can we just inline? Is `createResource` called several times?

Reply:
We can, I just copied the way it's done in the SDK.